src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBindingHelper.cs

```diff
@@ -748,46 +748,146 @@ internal static TModel CastOrDefault<TModel>(object model)
             return (model is TModel) ? (TModel)model : default(TModel);
         }
 
-        public static object ConvertValuesToCollectionType<T>(Type modelType, IList<T> values)
+        /// <summary>
+        /// Gets an indication whether <see cref="M:GetCompatibleCollection{T}"/> is likely to return a usable
+        /// non-<c>null</c> value.
+        /// </summary>
+        /// <typeparam name="T">The element type of the <see cref="ICollection{T}"/> required.</typeparam>
+        /// <param name="bindingContext">The <see cref="ModelBindingContext"/>.</param>
+        /// <returns>
+        /// <c>true</c> if <see cref="M:GetCompatibleCollection{T}"/> is likely to return a usable non-<c>null</c>
+        /// value; <c>false</c> otherwise.
+        /// </returns>
+        /// <remarks>"Usable" in this context means the property can be set or its value reused.</remarks>
+        public static bool CanGetCompatibleCollection<T>(ModelBindingContext bindingContext)
```

Comment:
@rynowak you might want to look at this code again or at the most recent commit. Simplified slightly and reduced block nesting.